### PR TITLE
Add new command !last_seen and make !seen case insensitive

### PR DESCRIPTION
**Pull request by quicquid**

### CODE REVIEW

File: src/core/Plugin_social.ml [hunk 1]
```diff
@@ -145,29 +145,62 @@ let print_diff (f:float) : string =
...
+         |> CCList.sort (fun a b -> - (compare (snd a) (snd b)) )
+         |> CCList.map (create_message_for_user now)
+         |> Lwt.return
        with e ->
          Lwt.fail (Command.Fail ("seen: " ^ Printexc.to_string e)))
 
+
+let cmd_lastseen (state:state) =
+  Command.make_simple_l
+    ~descr:"ask for the last time someone talked on this chan"
```
Comment (c-cube) [inline src/core/Plugin_social.ml]: shouldn't it be "last_talked", then? "seen" would be confusing otherwise

File: src/core/Plugin_social.ml [hunk 2]
```diff
@@ -145,29 +145,62 @@ let print_diff (f:float) : string =
...
-           | None ->
-             Lwt.return_some "who?"
-         end
+         let now = Unix.time () in
+         StrMap.fold (fun name data acc ->
+             if dest = String.uppercase name then
+               (name, data.last_seen) :: acc
+             else
+               acc
+           ) state.map []
```
Comment (c-cube) [inline src/core/Plugin_social.ml]: I'd rather move the `)` to the previous line

File: src/core/Plugin_social.ml [hunk 3]
```diff
@@ -145,29 +145,62 @@ let print_diff (f:float) : string =
...
+
 let cmd_seen (state:state) =
-  Command.make_simple
+  Command.make_simple_l
     ~descr:"ask for the last time someone talked on this chan"
     ~prio:10 ~prefix:"seen"
     (fun _msg s ->
        try
-         let dest = String.trim s in
+         let dest = String.trim s |> String.uppercase in
```
Comment (c-cube) [inline src/core/Plugin_social.ml]: `CCString.uppercase_ascii`, which works with unicode

File: src/core/Plugin_social.ml (hunk 1 above)
Comment (quicquid) [reply src/core/Plugin_social.ml]: What about just making it !last then?

**Pull request merged**